### PR TITLE
fix: [PIE-25770]: resizable and draggable support added

### PR DESCRIPTION
**Pull request by samarth-sharma31**
##### Summary:

Adds support for resizable and draggable components

##### Jira Links:

https://harness.atlassian.net/browse/CDS-25770

##### Screenshots:

https://user-images.githubusercontent.com/96409598/154804145-c40df51c-590d-4fc3-a8a8-5d1c76da8ef6.mov

https://user-images.githubusercontent.com/96409598/154804148-6aea4227-e6ee-4fe8-b2b4-0da295750deb.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/10-common/components/DraggableAndResizableWrapper/DraggableAndResizableWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      bounds="window"
+      enableResizing={{
+        top: true,
+        bottom: true,
+        left: true,
+        right: true,
+        topRight: false,
+        bottomRight: false,
+        bottomLeft: false,
+        topLeft: false
```
Comment (ishantmehtaharness) [inline src/modules/10-common/components/DraggableAndResizableWrapper/DraggableAndResizableWrapper.tsx:29]: Is it better to make these defaults and take as a prop as well?

**Review by kapilkumawat86 — CHANGES_REQUESTED** [review]
Use existing packages

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

File: package.json [hunk 2]
```diff
@@ -96,6 +96,7 @@
     "react-contenteditable": "^3.3.5",
     "react-dom": "^17.0.2",
     "react-draggable": "^4.4.2",
+    "react-rnd": "^10.3.5",
```
Comment (ishantmehtaharness) [inline package.json:99]: Not sure if this was discussed with team,
but as a good practice we should discuss this internally with team, before adding any new package to our existing codebase.

Comment (samarth-sharma31) [reply package.json:99]: yes sure, i'll sync with @chiragp1212 on this, @kapilkumawat86 had mentioned that he has used a similar library for this use case

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 24 Issues 
* Code issue 24 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/7404)

**Pull request closed without merging**